stm32-modules/tempdeck-gen3/firmware/thermal_control/thermal_hardware.c

```diff
@@ -153,15 +174,33 @@ bool thermal_hardware_set_peltier_cool(double power) {
     }
     
     uint32_t pwm = power * (double)MAX_PWM;
-    __HAL_TIM_SET_COMPARE(&hardware.timer, HEATING_CHANNEL, 0);
-    __HAL_TIM_SET_COMPARE(&hardware.timer, COOLING_CHANNEL, pwm);
+    __HAL_TIM_SET_COMPARE(&hardware.peltier_timer, HEATING_CHANNEL, 0);
+    __HAL_TIM_SET_COMPARE(&hardware.peltier_timer, COOLING_CHANNEL, pwm);
     
     hardware.hot_side_power = 0.0F;
     hardware.cool_side_power = power;
 
     return true;
 }
 
+bool thermal_hardware_set_fan_power(double power) {
+    if(!hardware.initialized) {
+        return false;
+    }
+    if(power > 1.0F) {
+        return false;
+    }
+    if(power == 0.0) {
+        // The fan controller will default to full power if it thinks the
+        // control line is disconnected, and unfortunately it thinks a 0% PWM
+        // is a disconnection. So the lowest allowable PWM is 0.01, which 
+        // still results in the fan staying still.
+        power = 0.01;
+    }
+    uint32_t pwm = power * (double)FAN_MAX_PWM;
```

Comment:
it makes me a little nervy to do this check this way, for two reasons:
- exact float comparison; what if somebody commands 0.005 power?
- what if for some reason in the future we change the period to a value not a multiple of 100, and this results in a 0 value in the compare register?

Recommend making the check more like
```
uint32_t pwm = power * (double)FAN_MAX_PWM;
if (pwm == 0) { pwm = 1; }
```

Reply:
@sfoster1 tested this change on hardware and it works 👍